ccl/backupccl: TestBackupRestorePartitioned failed

ccl/backupccl.TestBackupRestorePartitioned [failed](https://tanzanite.cluster.engflow.com/invocation/9f917b34-099d-49ca-aa75-b4acb6f02fad) on master @ [2859e5fab431536ec77f646787af637eec3d185c](https://github.com/cockroachdb/cockroach/commits/2859e5fab431536ec77f646787af637eec3d185c):


```
=== RUN   TestBackupRestorePartitioned
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupRestorePartitioned1056826626
    test_log_scope.go:81: use -show-logs to present logs inline
    testutils.go:289: no Invalid Descriptors
    backup_test.go:496: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupRestorePartitioned1056826626
--- FAIL: TestBackupRestorePartitioned (7.31s)
=== RUN   TestBackupRestorePartitioned/partition-by-different-tiers
    backup_test.go:316: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:316
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:472
        	Error:      	Should not be: ""
        	Test:       	TestBackupRestorePartitioned/partition-by-different-tiers
    --- FAIL: TestBackupRestorePartitioned/partition-by-different-tiers (0.63s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=2</code>
, <code>shard=22</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestorePartitioned.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33850

ccl/backupccl.TestBackupRestorePartitioned [failed](https://tanzanite.cluster.engflow.com/invocation/3ef5e249-f8d6-4daf-a31c-5c34f4c0fb69) on master @ [a87cefce46dc064e0d782dbfde55e0a1f3db1abc](https://github.com/cockroachdb/cockroach/commits/a87cefce46dc064e0d782dbfde55e0a1f3db1abc):


```
=== RUN   TestBackupRestorePartitioned
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupRestorePartitioned1204772566
    test_log_scope.go:81: use -show-logs to present logs inline
    testutils.go:289: no Invalid Descriptors
    backup_test.go:496: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupRestorePartitioned1204772566
--- FAIL: TestBackupRestorePartitioned (7.33s)
=== RUN   TestBackupRestorePartitioned/partition-by-different-tiers
    backup_test.go:316: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:316
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:472
        	Error:      	Should not be: ""
        	Test:       	TestBackupRestorePartitioned/partition-by-different-tiers
    --- FAIL: TestBackupRestorePartitioned/partition-by-different-tiers (1.88s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=16</code>
, <code>shard=22</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestorePartitioned.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupRestorePartitioned [failed](https://tanzanite.cluster.engflow.com/invocation/8dd72aed-8fea-46dd-8ce5-44c52fa58893) on master @ [caff15394fcbe37208b46b2973714c27cc3a1417](https://github.com/cockroachdb/cockroach/commits/caff15394fcbe37208b46b2973714c27cc3a1417):


```
=== RUN   TestBackupRestorePartitioned
initialized metamorphic constant "backup-presplit-spans" with value false
initialized metamorphic constant "export_request_verbose_tracing" with value true
initialized metamorphic constant "restore-worker-concurrency" with value 7
I231125 08:14:27.399437 1 (gostd) rand.go:250  [-] 1  random seed: 5546785664676446010
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupRestorePartitioned2300493474
    test_log_scope.go:81: use -show-logs to present logs inline
    testutils.go:289: no Invalid Descriptors
    backup_test.go:496: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupRestorePartitioned2300493474
--- FAIL: TestBackupRestorePartitioned (5.98s)
=== RUN   TestBackupRestorePartitioned/partition-by-different-tiers
    backup_test.go:316: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:316
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:472
        	Error:      	Should not be: ""
        	Test:       	TestBackupRestorePartitioned/partition-by-different-tiers
    --- FAIL: TestBackupRestorePartitioned/partition-by-different-tiers (0.74s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=25</code>
, <code>shard=22</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestorePartitioned.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -327,6 +327,9 @@ func requireHasNoSSTs(t *testing.T, locations ...string) {
 // are wrapped with SucceedsSoon() because EXPERIMENTAL_RELOCATE can fail if
 // there are other replication changes happening.
 func ensureLeaseholder(t *testing.T, sqlDB *sqlutils.SQLRunner) {
+	// Anything that is calling this to ensure *leaseholders* probably cares about
+	// the work going to the leaseholders as it did before follower reads.
+	sqlDB.Exec(t, `SET CLUSTER SETTING bulkio.backup.balanced_distribution.enabled = false`)
 	for _, stmt := range []string{
 		`ALTER TABLE data.bank SPLIT AT VALUES (0)`,
 		`ALTER TABLE data.bank SPLIT AT VALUES (100)`,

